soc/arm/st_stm32/stm32wl/Kconfig.soc

```diff
@@ -7,7 +7,21 @@ choice
 	prompt "STM32WLx MCU Selection"
 	depends on SOC_SERIES_STM32WLX
 
+config SOC_STM32WLE4XX
+	bool "STM32WLE4XX"
+
+config SOC_STM32WLE5XX
+	bool "STM32WLE5XX"
+
+config SOC_STM32WL54XX
+	bool "STM32WL54XX"
+
 config SOC_STM32WL55XX
 	bool "STM32WL55XX"
 
 endchoice
+
+config STM32WL_DUAL_CORE
```

Comment:
Is this switch supposed to be configured by user, or implicitly depending on MCU used ?

Knowing that there is already a CMSIS switch `DUAL_CORE ` (for example https://github.com/zephyrproject-rtos/hal_stm32/blob/d49fb6dbe780b931eee33cce8656e87c46070fa8/stm32cube/stm32wlxx/soc/stm32wl55xx.h#L43)
I think this switch is not needed, you can use:
`#if defined(CONFIG_SOC_SERIES_STM32WLX) && defined(DUAL_CORE )`

Reply:
Thanks for this suggestion. The intention was to do it similar as stm32h7 does it and enable this at board level.
H7 uses this flag in the flash driver to enable usage of hsem for locking, in the hsem driver itself, and in soc_m7 to start the other core after the kernel was initialized.

I have had the intention to use it to start the second core, but currently I am still awaiting the hardware to be able to test this so I left this out for now.
Would it be preferable to use DUAL_CORE flag from the hal to decide the clock configuration(and in the future for flash access). Additionally later introduce a new kconfig option to start the second core?

Reply:
Ok, on H7 side the difference between DUAL_CORE and STM32H7_DUAL_CORE is that 
* DUAL_CORE indicate that MCU is dualcore and it is directly set thanks to CMSIS based on MCU selection
*  STM32H7_DUAL_CORE indicated that Zephyr will manage the dualcore 
 (if not set, zephyr will run as a single core even if MCU is dualcore)

So I agree to mimic what is done on H7 , and keep switch CONFIG_SOC_SERIES_STM32WLX configurable by user.
But Introducing this switch suppose that zephyr is able to handle STM32WL dualcore ... which is not the case for now.  
It would require to add 
* specific M4 vs M0 configuration, DeviceTree, yaml, ...
In other world: be able to compile zephyr for M4 and for M0 (currently only M4 is supported)
* dualcore boot:  for example boot 1st on M4 which will configure the clock, while M0 is stopped, then M4 wakeup M0 , and M0 do not configure the clock ...
* lowpower enter/exit : how to manage the clock
* manage Hardware semaphore everywhere it is necessary (shared IP like RCC, PWR, ...)
* ...

Reply:
Yes agree.
I have removed the stm32wl dual core configuration option, as at this point it does not yet make sense.

Reply:
Sorry to be late, but I thing we should avoid direct use of `DUAL_CORE` for the reason it is unclear and easily misleading. Its usage is fine inside CubeWL, but outside (and here in zephyr) it could lead to misunderstanding as it lacks basic namespacing.
So, to the minimum, I suggest we introduce a dedicated macro coming with some context explanation.
This will work for this driver, but if we need this across drivers I'm not sure we'll be able to avoid defining a Kconfig symbol as suggested initially.

Reply:
Following the suggestion to avoid direct usage of DUAL_CORE, I am now using:

```
/* Identify stm32wl dual-core socs by symbol defined in CMSIS dev header file */
#if (defined(CONFIG_SOC_SERIES_STM32WLX) && defined(DUAL_CORE))
#define STM32WL_DUAL_CORE
#endif
```

And we can introduce a Kconfig symbol, once better better dual core support is available.
Any alternative suggestions?